### PR TITLE
Define vlan interfaces to be included in introspection report

### PR DESCRIPTION
**Pull request by bfournie**
Define the inspector kernel parameter to include vlan interfaces
and associated IP addresses in the introspection report.

### CODE REVIEW

**Comment by hardys** [comment]
Thanks @bfournie - does this depend on a corresponding change to ironic-image or ironic-inspector-image?

**Comment by bfournie** [comment]
> Thanks @bfournie - does this depend on a corresponding change to ironic-image or ironic-inspector-image?

Yes, https://github.com/metal3-io/ironic-image/pull/246

**Comment by dhellmann** [comment]
How does the VLAN data show up? Are there just more interfaces added to the existing data structure, or do we need to do something to expand the API to hold new data?

**Comment by bfournie** [comment]
> How does the VLAN data show up? Are there just more interfaces added to the existing data structure, or do we need to do something to expand the API to hold new data?

Yes, there are just more interfaces added to the existing structure so no expansion is needed.  The vlan interfaces will be named <interface.vlan> e.g. "eth0.100" and will be handled as normal interfaces.

**Comment by dhellmann** [comment]
> > How does the VLAN data show up? Are there just more interfaces added to the existing data structure, or do we need to do something to expand the API to hold new data?
> 
> Yes, there are just more interfaces added to the existing structure so no expansion is needed. The vlan interfaces will be named <interface.vlan> e.g. "eth0.100" and will be handled as normal interfaces.

Sounds good, thanks.

**Comment by dtantsur** [comment]
/approve

The code looks good, but you need to fix markdown (I suspect redundant indentation).

File: docs/dev-setup.md [hunk 1]
```diff
@@ -80,6 +80,10 @@ The following environment variables can be passed to configure the ironic:
 - IRONIC_FAST_TRACK - whether to enable fast_track provisioning or not
   (default true)
 - IRONIC_KERNEL_PARAMS - Kernel parameters to pass to IPA (default console=ttyS0)
+- IRONIC_INSPECTOR_VLAN_INTERFACES - VLAN interfaces to be included in introspection report
+       `all` - all VLANs on all interfaces, using LLDP information (default)
+       interface - all VLANs on an interface, using LLDP information
+       interface.vlan - a particular VLAN interface, not using LLDP
```
Comment (dtantsur) [inline docs/dev-setup.md:86]: Newlines are not rendered in markdown, you need at least commas between the items

**Review by elfosardo — APPROVED** [review]

**Comment by hardys** [comment]
lgtm but needs a rebase @bfournie

**Review by iurygregory — APPROVED** [review]

**Pull request merged**